xactserver/proto/rpc.proto

```diff
@@ -11,15 +11,22 @@ message PrepareRequest {
 message PrepareResponse {
 }
 
-enum Vote {
-    COMMIT = 0;
-    ABORT = 1;
-}
-
 message VoteRequest {
     uint32 from = 1;
     uint64 xact_id = 2;
-    Vote vote = 3;
+    // If this field is not set, this is a 'yes' vote
+    oneof rollback_reason {
+        DbError db = 3;
+        string other = 4;
```

Comment:
Why? Shouldn't we just go into the detail part of DbError. Or do we expect to abort without an error?

Reply:
These fields store the errors returned from the `tokio_postgres` library. It has 2 types of errors: standardized SQL errors (e.g. deadlock) and non-SQL errors (e.g. connection closed). The "other" field here may also store non-SQL errors at our level of abstraction.

Reply:
Why not use the "Class 08 — Connection Exception" for the connection related errors?

Reply:
I think that one is for connection inside postgres (e.g. a follower connecting to the master). If the error is because the postgres client cannot connect to postgres then postgres cannot know about it.